### ISSUE TITLE
[Bug]: After killing querynode and waiting 10 minutes, get replicas return the collection not loaded or leader is offline[NodeNotFound(0)]

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#22711]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.0-20230310-b2ece6a5
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

1. Deploy a cluster with 4 query nodes
2. Create `RG_0` and transfer 1 query node to it, create `RG_1` and transfer 1 query node to it
3. Load 2 replicas of collection `ResourceGroup_111`  into resource groups `['RG_0', 'RG_1']`
4. Load 2 replicas of collection `ResourceGroup_222` into default_rg
5. Chaos pod-kill query node multi times
6. Sleep 10 minutes
7. Get replicas of `ResourceGroup_222` and get exceptions
```
message=failed to get replica info, err=failed to get shard leader for shard collectionID:440046547801935909 channelName:"by-dev-rootcoord-dml_7_440046547801935909v1" seek_position:<channel_name:"by-dev-rootcoord-dml_7_440046547801935909v1" msgID:"\010\t\020H\030\000 \000" msgGroup:"by-dev-dataNode-9-by-dev-rootcoord-dml_7_440046547801935909v1" timestamp:440046598284115969 > flushedSegmentIds:440046547801935921 , the collection not loaded or leader is offline[NodeNotFound(0)])>
```
And there are only one replica is loaded of colelction `ResourceGroup_222`:
```
[2023-03-12T18:25:18.009Z] [2023-03-12 18:25:17 - DEBUG - ci_test]: (api_response) : [segmentID: 440046547801935921

[2023-03-12T18:25:18.009Z] collectionID: 440046547801935909

[2023-03-12T18:25:18.009Z] partitionID: 440046547801935910

[2023-03-12T18:25:18.009Z] num_rows: 5035

[2023-03-12T18:25:18.009Z] state: Sealed

[2023-03-12T18:25:18.009Z] nodeIds: 13

[2023-03-12T18:25:18.009Z] , segmentID: 440046547801935920

[2023-03-12T18:25:18.009Z] collectionID: 440046547801935909

[2023-03-12T18:25:18.009Z] partitionID: 440046547801935910

[2023-03-12T18:25:18.009Z] num_rows: 4965

[2023-03-12T18:25:18.009Z] state: Sealed

[2023-03-12T18:25:18.009Z] nodeIds: 17

[2023-03-12T18:25:18.009Z] nodeIds: 18

[2023-03-12T18:25:18.009Z] ]  (api_request.py:31)
```

The question is why the replicas have not been loaded in 10 minutes ?

### Expected Behavior

Collections are all loaded after recovering from chaos

### Steps To Reproduce

```markdown
Jenkins pipeline: chaos test -> chaos-test-resource-group
```

### Milvus Log

jenkins job:  https://qa-jenkins.milvus.io/blue/organizations/jenkins/chaos-test-resource-group/detail/chaos-test-resource-group/143/pipeline/313
devops cluster, chaos-testing namespace
```
[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-datacoord-847f9f44f8-c8rg4          1/1     Running             0                19m     10.102.7.193    devops-node11   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-datanode-5b878696c7-49wwx           1/1     Running             0                19m     10.102.5.77     devops-node21   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-datanode-5b878696c7-r5w6k           1/1     Running             0                19m     10.102.10.57    devops-node20   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-indexcoord-57f47f9cd9-64klk         1/1     Running             0                19m     10.102.10.58    devops-node20   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-indexnode-558c8bc8fb-swpvh          1/1     Running             0                19m     10.102.10.61    devops-node20   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-proxy-587898fbc5-4sxts              1/1     Running             0                19m     10.102.10.49    devops-node20   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-querycoord-6fc56f9478-42g4j         1/1     Running             0                19m     10.102.5.75     devops-node21   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-querynode-67f66997bb-7kz6c          1/1     Running             0                12m     10.102.10.109   devops-node20   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-querynode-67f66997bb-7zk25          1/1     Running             0                12m     10.102.7.39     devops-node11   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-querynode-67f66997bb-8vdkz          1/1     Running             0                13m     10.102.5.150    devops-node21   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-querynode-67f66997bb-dlrv6          1/1     Running             0                19m     10.102.9.209    devops-node13   <none>           <none>

[2023-03-12T18:24:10.747Z] querynode-pod-kill-143-milvus-rootcoord-848dcb5668-x49ft          1/1     Running             0                19m     10.102.10.63    devops-node20   <none>           <none>
```

**Comment by weiliu1031** [comment]
here comes two issue:
1. after many times kill-querynode-chaos, all available querynode has been assign to one replica, and there are two replica in `ResourceGroup_222 `, so one replica has no available query node. which cause by we see nodeUp before nodeDown when querynode restarts, so when assign node to replica, we don't know the real available node num in each replica.
2. the replica with two querynode also meet load problems `get data distribution failed caused by context exceed`, as a known issue, @congqixia is working on it

**Comment by weiliu1031** [comment]
should be fixed by #23626, please verify this
